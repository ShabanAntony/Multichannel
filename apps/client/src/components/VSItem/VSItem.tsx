import React, { FC, memo } from 'react'

import { SaleItemEntity } from '@server-api'
import classNames from 'classnames'

import { laryToUSD } from '../../constants/prices'
import {
  calculateVSFinalPrice,
  formatLaryPrice,
  formatUSDPrice,
} from '../../utils/price'
import { ImageLink } from '../ImageLink/ImageLink'

import styles from './VSItem.module.scss'

export interface Props {
  readonly data: SaleItemEntity
}

export const VSItem: FC<Props> = memo(function VSItem({
  data: {
    name,
    url,
    parsedPrice,
    parsedDiscount,
    discount,
    productImages,
    swatches,
    isNew,
  },
}) {
  const finalPrice = calculateVSFinalPrice(parsedDiscount)
  const baseClasses = classNames(styles.base, {
    [styles.base__new]: isNew,
  })

  return (
    <li className={baseClasses}>
      <ImageLink
        className={styles.image}
        link={`${process.env.NX_PUBLIC_VS_HOST}${url}`}
        src={`${process.env.NX_PUBLIC_VS_HOST}/p/200x280/${productImages[0]}.jpg`}
        alt={name}
        width={200}
        height={280}
      />
      <p className={styles.name}>{name}</p>
      <p>
        Prices: {parsedPrice}
        {' > '}
        <span className={styles.discount}>
          {parsedDiscount} ({Math.round(discount)}%)
        </span>
        {' > '}
        <span className={styles.finalPrice}>
          {formatUSDPrice(finalPrice)} (
          {formatLaryPrice(finalPrice * laryToUSD)})
        </span>
      </p>
      <div className={styles.swatches}>
        {swatches
          .filter(({ swatchImage }) => !!swatchImage)
          .map(({ url, swatchImage, name }, index) => (
            <ImageLink
              className={styles.swatch_link}
              key={`swatch-${name}-${index}`}
              link={`${process.env.NX_PUBLIC_VS_HOST}${url}`}
              src={`${process.env.NX_PUBLIC_VS_HOST}/p/125x125/${swatchImage}.jpg`}
              alt={name}
              width={35}
              height={35}
            />
          ))}
      </div>
    </li>
  )
})
