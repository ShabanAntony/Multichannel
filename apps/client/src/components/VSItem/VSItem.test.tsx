import React from 'react'
import renderer from 'react-test-renderer'

import { SaleItemEntity } from '@server-api'

import { VSItem } from './VSItem'

jest.mock('../ImageLink/ImageLink', () => ({
  ImageLink: 'ImageLink',
}))

const mockItem: SaleItemEntity = {
  discount: 0,
  isNew: false,
  masterStyleId: '',
  name: '',
  parsedDiscount: 0,
  parsedPrice: 0,
  price: '',
  productImages: ['image'],
  regDate: '',
  salePrice: '',
  swatches: [
    {
      productImage: 'productImage',
      family: 'family',
      name: 'name',
      url: 'url',
    },
  ],
  url: '',
}

describe('VSItem', () => {
  it('should renders regular item', () => {
    const tree = renderer.create(<VSItem data={mockItem} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should renders new item', () => {
    const tree = renderer
      .create(<VSItem data={{ ...mockItem, isNew: true }} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
