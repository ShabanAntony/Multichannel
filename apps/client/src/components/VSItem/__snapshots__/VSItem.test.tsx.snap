// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VSItem should renders new item 1`] = `
<li
  className="base base__new"
>
  <ImageLink
    alt=""
    className="image"
    height={280}
    link="http://victoriassecret.com"
    src="http://victoriassecret.com/p/200x280/image.jpg"
    width={200}
  />
  <p
    className="name"
  />
  <p>
    Prices: 
    0
     &gt; 
    <span
      className="discount"
    >
      0
       (
      0
      %)
    </span>
     &gt; 
    <span
      className="finalPrice"
    >
      0$
       (
      0 ლ
      )
    </span>
  </p>
  <div
    className="swatches"
  />
</li>
`;

exports[`VSItem should renders regular item 1`] = `
<li
  className="base"
>
  <ImageLink
    alt=""
    className="image"
    height={280}
    link="http://victoriassecret.com"
    src="http://victoriassecret.com/p/200x280/image.jpg"
    width={200}
  />
  <p
    className="name"
  />
  <p>
    Prices: 
    0
     &gt; 
    <span
      className="discount"
    >
      0
       (
      0
      %)
    </span>
     &gt; 
    <span
      className="finalPrice"
    >
      0$
       (
      0 ლ
      )
    </span>
  </p>
  <div
    className="swatches"
  />
</li>
`;
